### PR TITLE
Add some tests and some final fixes/updates

### PR DESCRIPTION
**Pull request by GregSutcliffe**
More tests to come ;)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ right now ;)
 Require the gem in Foreman.
 
     bundler.d/Gemfile.local.rb:
-    gem 'foreman_discovery', :git => "https://github.com/GregSutcliffe/foreman_discovery.git"
+    gem 'foreman_discovery', :git => "https://github.com/theforeman/foreman_discovery.git"
```
Comment (ohadlevy) [inline README.md:19]: I would prefer the default location would be an official Gem?

File: app/controllers/discovers_controller.rb [hunk 2]
```diff
@@ -13,11 +13,12 @@ class DiscoversController < ::ApplicationController
...
-    # sometimes loses the `where` clause, leading to all Hosts on the Discovery index
-    hosts.reject! {|h| h.type != "Host::Discovered" }
     respond_to do |format|
-      format.html { @hosts = hosts.includes(:model, :location, :organization).paginate :page => params[:page] }
+      format.html do
+        # This `where` shouldn't be necessary, but randomly the sql query on the line above
+        # sometimes loses the `where` clause, leading to all Hosts on the Discovery index
+        @hosts = hosts.where(:type => "Host::Discovered").includes(:model, :location, :organization).paginate :page => params[:page]
+      end
       format.json { render :json => hosts }
```
Comment (ohadlevy) [inline app/controllers/discovers_controller.rb:21]: either make sure the json response include the same type, or remove json all together imho

File: README.md (hunk 1 above)
Comment (GregSutcliffe) [inline README.md:19]: I agree, we can release one later today. This is correct for running the 'latest' version, so we'll extend it when the gem goes out.

File: app/controllers/discovers_controller.rb (hunk 2 above)
Comment (GregSutcliffe) [inline app/controllers/discovers_controller.rb:21]: Fix incoming

**Pull request merged**